jekyll/assets/js/sidebar.js

```diff
@@ -0,0 +1,45 @@
+// making sidebar stick when touching footer
+// this improves the visual experience while interacting with the docs site
+(function () {
+  var footer = document.querySelector('.footer');
+  var sidebar = document.querySelector('.sidebar');
+
+  // get hash, if it exists
+  if (window.location.hash && window.location.hash.indexOf('section') > -1) {
+    var section = getUrlVars(window.location.hash);
+    localStorage.sidenavActive = section['section']
+  }
+  if (localStorage.sidenavActive) {
+    var element = sidebar.querySelector('[data-section=' + localStorage.sidenavActive + ']');
+    if (element.classList.contains('closed')) {
+      element.classList.remove('closed');
+    }
+  }
+
+  window.addEventListener('scroll', function () {
+    // if footer is in frame, removed fixed style (otherwise add it, if it doesn't exist)
+    if ((footer.getBoundingClientRect().top - window.innerHeight) <= 0) {
```

Comment:
I'm concerned about running this within a scroll event callback. I believe both `getBoundingClientRect` and `innerHeight` will cause a paint/render, which means that the page will have to paint/render on each individual scroll event (of which there are many), causing performance issues. Maybe the easiest solution is just to [`debounce`](https://lodash.com/docs/4.17.4#debounce) this callback function so it only gets called after the user stops scrolling?

Reply:
Yeah, maybe we can pair on this today. I've run it locally, and I think you're right that it may not be a concern, but it's worth chatting about, I think.